From 81ba0ec901e265d1bdfdf4c9b0bf46d12775845b Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Thu, 26 Oct 2017 16:30:24 +0800
Subject: [PATCH] drm: bridge: dw-hdmi: add switch state in
 dw_hdmi_connector_force

Application need to listen HDMI connector state when connector is
forced on/off, so we add switch_set_stat in dw_hdmi_connector_force.

Change-Id: I2b76a0a647eb6a4cfde7584e085f53540d0fa27f
Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/gpu/drm/bridge/dw-hdmi.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/bridge/dw-hdmi.c b/drivers/gpu/drm/bridge/dw-hdmi.c
index 5156271b0660..59d6ac609794 100644
--- a/drivers/gpu/drm/bridge/dw-hdmi.c
+++ b/drivers/gpu/drm/bridge/dw-hdmi.c
@@ -2358,6 +2358,14 @@ static void dw_hdmi_connector_force(struct drm_connector *connector)
 					     connector);
 
 	mutex_lock(&hdmi->mutex);
+#ifdef CONFIG_SWITCH
+	if (!hdmi->disabled && hdmi->force != connector->force) {
+		if (connector->force == DRM_FORCE_OFF)
+			switch_set_state(&hdmi->switchdev, 0);
+		else if (connector->force == DRM_FORCE_ON)
+			switch_set_state(&hdmi->switchdev, 1);
+	}
+#endif
 	hdmi->force = connector->force;
 	dw_hdmi_update_power(hdmi);
 	dw_hdmi_update_phy_mask(hdmi);
-- 
2.35.3

